src/object/cli_shard.c

```diff
@@ -896,6 +896,9 @@ dc_rw_cb(tse_task_t *task, void *arg)
 		D_GOTO(out, rc = -DER_HG);
 	}
 
+	reasb_req = rw_args->shard_args->reasb_req;
+	is_ec_obj = reasb_req != NULL && daos_oclass_is_ec(reasb_req->orr_oca);
```

Comment:
probably not this patch issue, but looks like reasb_req != NULL is enough here?

Reply:
Just move the logic from other place, but it seems yes.

Reply:
"reasb_req != NULL" looks enough, and probably better to use obj_is_ec(obj) and assert(reasb_req != NULL) for ec obj. the old original code probably just not sure if need the reasb_req for replica at that time.